extends Node

@onready var menu_principal = $Control
@onready var tela_de_opÃ§oes = $"TelaDeOpÃ§oes"
@onready var musica = $"TelaDeOpÃ§oes/MarginContainer/VBoxContainer/musica"
var musica_mutada = false


func _on_opÃ§Ãµes_pressed() -> void:
	menu_principal.visible = false


func _on_menu_principal_pressed() -> void:
	menu_principal.visible = true


func _on_musica_pressed() -> void:
	musica_mutada = !musica_mutada
	
	if musica_mutada:
		AudioServer.set_bus_volume_db(AudioServer.get_bus_index("Master"), -80)
	
	else:
		AudioServer.set_bus_volume_db(AudioServer.get_bus_index("Master"), 0)
		
	atualizar_texto_botao()
	
func atualizar_texto_botao():
	if musica_mutada:
		$"TelaDeOpÃ§oes/MarginContainer/VBoxContainer/musica".text = "ðŸ”‡ MÃºsica: OFF"
	else:
		$"TelaDeOpÃ§oes/MarginContainer/VBoxContainer/musica".text = "ðŸ”Š MÃºsica: ON"


		
